src/profile_manager/views.py

```diff
@@ -59,19 +77,36 @@ class ProfileListCurrent(ProfileList):
     Arguments:
         ProfileList -- Base class
     """
+    view_type = ViewTypes.CURRENT
 
     # override the staff requirement for ProfileList
     def test_func(self):
         return self.request.user.is_authenticated
 
     def get_queryset(self):
-        profiles_qs = Profile.objects.all_for_active_semester()
+        profiles_qs = Profile.objects.all_for_active_semester().filter(user__is_active=True)
         return self.queryset_append(profiles_qs)
 
-    def get_context_data(self, **kwargs):
-        context = super(ProfileListCurrent, self).get_context_data(**kwargs)
-        context['current_only'] = True
-        return context
+
+@method_decorator(staff_member_required, name='dispatch')
+class ProfileListStaff(ProfileList):
+    view_type = ViewTypes.STAFF
+
+    def get_queryset(self):
+        profiles_qs = Profile.objects.filter(user__is_staff=True)
+        return self.queryset_append(profiles_qs)
+
+
+class ProfileListInactive(ProfileList):
+    view_type = ViewTypes.INACTIVE
+
+    # override the staff requirement for ProfileList
+    def test_func(self):
+        return self.request.user.is_authenticated
+
+    def get_queryset(self):
+        profiles_qs = Profile.objects.all_students().filter(user__is_active=False)
+        return self.queryset_append(profiles_qs)
```

Comment:
why append and not just return profile_qs? What does the qs contain at this point that you're appending to? (same question for above get_queryset() methods

Reply:
removed for both